build/lib/dome9ApiV2Py/dome9ApiV2Py.py

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env python
+import json
+import requests
+from requests import ConnectionError, auth
+import urlparse
+
+class Dome9ApiSDK(object):
+	REGION_PROTECTION_MODES = ['FullManage', 'ReadOnly', 'Reset']
+	SEC_GRP_PROTECTION_MODES = ['FullManage', 'ReadOnly']
+	
+	def __init__(self, apiKeyID, apiSecret, apiAddress='https://api.dome9.com', apiVersion='v2'):
+		self.apiKeyID = apiKeyID
+		self.apiSecret = apiSecret
+		self.apiAddress = apiAddress
+		self.apiVersion = '/{}/'.format(apiVersion)
+		self.baseAddress = self.apiAddress + self.apiVersion
+		self.clientAuth = auth.HTTPBasicAuth(self.apiKeyID, self.apiSecret)
+		self.restHeaders = {'Accept': 'application/json', 'Content-Type': 'application/json'}
+		if not self.apiKeyID or not self.apiSecret:
+			raise Exception('Cannot create api client instance without keyID and secret!')
+
+# System methods
+	def get(self, route, payload=None):
+		return self.request('get', route, payload)
+
+	def post(self, route, payload=None):
+		return self.request('post', route, payload)
+
+	def patch(self, route, payload=None):
+		return self.request('patch', route, payload)
+
+	def put(self, route, payload=None):
+		return self.request('put', route, payload)
+
+	def delete(self, route, payload=None):
+		return self.request('delete', route, payload)
+
+	def request(self, method, route, payload=None, isV2=True):
+		res = None
+		url = None
+		try:
+			url = urlparse.urljoin(self.baseAddress, route)
+			if method == 'get':
+				res = requests.get(url=url, params=payload, headers=self.restHeaders, auth=self.clientAuth)
+
+			elif method == 'post':
+				res = requests.post(url=url, data=payload, headers=self.restHeaders, auth=self.clientAuth)
+
+			elif method == 'patch':
+				res = requests.patch(url=url, json=payload, headers=self.restHeaders, auth=self.clientAuth)
+
+			elif method == 'put':
+				res = requests.put(url=url, data=payload, headers=self.restHeaders, auth=self.clientAuth)
+
+			elif method == 'delete':
+				res = requests.delete(url=url, params=payload, headers=self.restHeaders, auth=self.clientAuth)
+
+		except requests.ConnectionError as ex:
+			raise ConnectionError(url, ex.message)
+
+		jsonObject = None
+		err = None
+
+		if res.status_code in range(200, 299):
+			try:
+				if res.content:
```

Comment:
This is bad
The use in inner try;except insise if
why would the try scope get exception?